server/lib/tuist_web/authentication.ex

```diff
@@ -340,7 +340,12 @@ defmodule TuistWeb.Authentication do
         require_authenticated_user(conn, opts)
 
       {:ok, preview} ->
-        if Authorization.can(nil, :read, Repo.preload(preview, :project)) do
+        if preview.visibility == :public or
```

Comment:
soo ... this piece somewhat sucks – but the preview is a bit of an odd apple where it doesn't align with our rules of `:project_preview_read` and passing the project entity over passing the preview. I'd rather keep consistent with the new declarative approach and keep this little piece of authorization here – or would you prefer to have `:preview_read` where the entity passed would be the `preview` over `project`?